main.js

```diff
@@ -31,3 +39,14 @@ function renderPhrase(currentState) {
 }
 
 $phraseContainer.appendChild(renderPhrase(applicationState))
+
+$window.addEventListener('keydown', function (event) {
+  if (event.key !== applicationState.characters[applicationState.currentIndex].char){
```

Comment:
It would be clearer to declare a variable instead of having a large accessor expression.

```js
var currentCharacter = applicationState.characters[applicatonState.currentIndex]
if (event.key !== currentCharacter.char) {
  currentCharacter.failure += 1
}
```